storage/innobase/dict/dict0mem.cc

```diff
@@ -303,6 +303,16 @@ dict_mem_table_add_col(
 	col = dict_table_get_nth_col(table, i);
 
 	dict_mem_fill_column_struct(col, i, mtype, prtype, len);
+
+	if (prtype & DATA_VERS_ROW_START) {
+		ut_ad(table->flags2 & DICT_TF2_VERSIONED
+			&& !(prtype & DATA_VERS_ROW_END));
+		table->vers_row_start = i;
+	} else if (prtype & DATA_VERS_ROW_END) {
+		ut_ad(table->flags2 & DICT_TF2_VERSIONED
+			&& !(prtype & DATA_VERS_ROW_START));
+		table->vers_row_end = i;
+	}
```

Comment:
I guess no, this doesn't work with `ROW_FORMAT=REDUNDANT`.

Reply:
If we did not add anything to `dict_table_t::flags2`, then I think it should work for any table, because no columns in `SYS_COLUMNS` ever contained uninitialized garbage. The `dict_table_t::flags2` change is kind-of redundant, because setting the new flags in `SYS_COLUMNS` already implies that system versioning is being used.